### PR TITLE
DOC: Fix docstring for events in solve_ivp

### LINKED ISSUE #9666: DOC: scipy.integrate.solve_ivp
> You can assign attributes like event.terminal = True to any function in Python. If None (default), events won’t be tracked.

That's very confusing.    It's not the attribute **event.terminal** which is set to None for no tracking, but the argument **events**.

### PR DESCRIPTION
**Pull request by MatthewFlamm**
Fixes #9666 . Moves default of events description to beginning rather than after the discussion of events.terminal.

### CODE REVIEW

File: scipy/integrate/_ivp/ivp.py [hunk 1]
```diff
@@ -234,13 +234,14 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
-        must have the signature ``event(t, y)`` and return a float. The solver will
-        find an accurate value of ``t`` at which ``event(t, y(t)) = 0`` using a
-        root-finding algorithm. Additionally each ``event`` function might have
-        the following attributes:
+    events : callable, or list of callables, optional
+        Types of events to track. If None (default), events won't be tracked.
+        Each is defined by a continuous function of time and state that becomes
+        zero value in case of an event. Each function must have the signature
+        ``event(t, y)`` and return a float. The solver will find an accurate
+        value of ``t`` at which ``event(t, y(t)) = 0`` using a root-finding
```
Comment (Kai-Striega) [inline scipy/integrate/_ivp/ivp.py]: Very minor issue, the double backticks around t, should only be a single backtick.

**Comment by MatthewFlamm** [comment]
Found quite a few double backticks to change.  I think there needs to be a more comprehensive review of backtick usage in this docstring.  Sometimes None has backticks, sometimes not.  But I think I fixed the obvious cases where double backticks should be single.

**Comment by Kai-Striega** [comment]
@MatthewFlamm it's looking good, the failing tests aren't related to your changes. #9736 should fix them, once that's been merged could you rebase and push it again, just to make sure?
Thank you for being so patient with this.

**Comment by MatthewFlamm** [comment]
@Kai-Striega Finally passed.

File: scipy/integrate/_ivp/ivp.py [hunk 2]
```diff
@@ -234,13 +234,14 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
-        Types of events to track. Each is defined by a continuous function of
-        time and state that becomes zero value in case of an event. Each function
-        must have the signature ``event(t, y)`` and return a float. The solver will
-        find an accurate value of ``t`` at which ``event(t, y(t)) = 0`` using a
-        root-finding algorithm. Additionally each ``event`` function might have
-        the following attributes:
+    events : callable, or list of callables, optional
+        Types of events to track. If None (default), events won't be tracked.
+        Each is defined by a continuous function of time and state that becomes
+        zero value in case of an event. Each function must have the signature
```
[the remark this answers is not in the document]
Comment (MatthewFlamm) [reply scipy/integrate/_ivp/ivp.py]: I personally think both versions are too verbose. I'd suggest the following instead "_Each event occurs at the zeros of a continuous function of time and state._".  Descriptions of the function return and inputs are detailed after this sentence, so I think this sentence can be shorter and simpler.

I'm not sure that the function truly needs to be continuous, just continuous near the zero crossings and sign changes.  This is probably too fine a point however and hard to write succintly and without confusing.

[the remark this answers is not in the document]
Comment (Kai-Striega) [reply scipy/integrate/_ivp/ivp.py]: Your version sounds better than mine. A quick clarification, do events occur at the *first* zero of the interval or for *all* zeros of the function?

File: scipy/integrate/_ivp/ivp.py [hunk 3]
```diff
@@ -234,13 +234,14 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
         within `t_span`. If None (default), use points selected by the solver.
-    events : callable, list of callables or None, optional
-        Types of events to track. Each is defined by a continuous function of
-        time and state that becomes zero value in case of an event. Each function
-        must have the signature ``event(t, y)`` and return a float. The solver will
-        find an accurate value of ``t`` at which ``event(t, y(t)) = 0`` using a
-        root-finding algorithm. Additionally each ``event`` function might have
-        the following attributes:
+    events : callable, or list of callables, optional
+        Types of events to track. If None (default), events won't be tracked.
```
Comment (Kai-Striega) [inline scipy/integrate/_ivp/ivp.py]: ``Types of events to track``: Are these the *types* of the events or the events to track?

``event's won't be tracked`` -> ``no events will be tracked.``: This is largely personal preference, I just think it sounds better.

**Comment by Kai-Striega** [comment]
@MatthewFlamm I agree your version sounds better and clearer.

**Comment by MatthewFlamm** [comment]
I think we've cleaned this up quite a bit.  I removed `Type of...` from the beginning of the docstring.  Your second suggestion I am +0.05 on, but I changed that too.

I also changed the formatting of the attributes.  From my inspection of other docstrings, it seems like the bullet point list is used to denote different possible options for an input, like under the `methods` input in `solve_ivp`.  In other docstrings, indented inputs follow the same pattern as a normal input.  See constraint input to [minimize](https://github.com/scipy/scipy/blob/722bfc3f2cb4884c7ef3eb87cefcb91ff7479208/scipy/optimize/_minimize.py#L146-L169).  This also helps to explicitly highlight the optional nature of these attributes, which was also added.

Minimize suffers from similar problems of `{type1, type2, type2}`, but it has the pattern `{type1, str1, str2, bool}`, which I wouldn't know what to do with.

**Comment by Kai-Striega** [comment]
@MatthewFlamm sorry I've missed your response. I think your changes are good and covers the scope of #9666. I'm happy to merge this if you don't have any extra changes in mind.

Regarding the formatting in minimize, I would add this into a new issue.

**Comment by MatthewFlamm** [comment]
I don't have any more changes now.  Thanks @Kai-Striega

**Comment by Kai-Striega** [comment]
Merged 👍. Thank you for taking the time to contribute.

**Pull request merged**